wallet/src/signer/trezor_signer/mod.rs

```diff
@@ -0,0 +1,1114 @@
+            set_value(value, &mut out_req);
+            out_req.set_address(
+                Address::new(chain_config, dest.clone()).expect("addressable").into_string(),
+            );
+
+            let mut out = MintlayerTxOutput::new();
+            out.transfer = Some(out_req).into();
+            out
+        }
+        TxOutput::LockThenTransfer(value, dest, lock) => {
+            let mut out_req = MintlayerLockThenTransferTxOutput::new();
+            match value {
+                OutputValue::Coin(amount) => {
+                    let mut value = MintlayerOutputValue::new();
+                    value.set_amount(amount.into_atoms().to_be_bytes().to_vec());
+                    out_req.value = Some(value).into();
+                }
+                OutputValue::TokenV1(token_id, amount) => {
+                    let mut value = MintlayerOutputValue::new();
+                    value.set_amount(amount.into_atoms().to_be_bytes().to_vec());
+                    value.set_token_id(token_id.to_hash().as_bytes().to_vec());
+                    out_req.value = Some(value).into();
+                }
+                OutputValue::TokenV0(_) => {
+                    panic!("token v0 unsuported");
+                }
+            };
+            out_req.set_address(
+                Address::new(chain_config, dest.clone()).expect("addressable").into_string(),
+            );
+
+            let mut lock_req = trezor_client::protos::MintlayerOutputTimeLock::new();
+            match lock {
+                OutputTimeLock::UntilTime(time) => {
+                    lock_req.set_until_time(time.as_int_seconds());
+                }
+                OutputTimeLock::UntilHeight(height) => {
+                    lock_req.set_until_height(height.into_int());
+                }
+                OutputTimeLock::ForSeconds(sec) => {
+                    lock_req.set_for_seconds(*sec);
+                }
+                OutputTimeLock::ForBlockCount(count) => {
+                    lock_req.set_for_block_count(*count);
+                }
+            }
+            out_req.lock = Some(lock_req).into();
+
+            let mut out = MintlayerTxOutput::new();
+            out.lock_then_transfer = Some(out_req).into();
+            out
+        }
+        TxOutput::Burn(value) => {
+            let mut out_req = MintlayerBurnTxOutput::new();
+            match value {
+                OutputValue::Coin(amount) => {
+                    let mut value = MintlayerOutputValue::new();
+                    value.set_amount(amount.into_atoms().to_be_bytes().to_vec());
+                    out_req.value = Some(value).into();
+                }
+                OutputValue::TokenV1(token_id, amount) => {
+                    let mut value = MintlayerOutputValue::new();
+                    value.set_amount(amount.into_atoms().to_be_bytes().to_vec());
+                    value.set_token_id(token_id.to_hash().as_bytes().to_vec());
```

Comment:
this looks out of sync with firmware code. I can't build it with pointing to `feature/mintlayer-pk` branch either